### PR TITLE
Adding env. vars. AWS_ACCESS_KEY and AWS_SECRET_ACCESS_KEY to installation files (#49).

### PR DESCRIPTION
**Pull request by mjkrause**
I added the two environment variables AWS_ACCESS_KEY and AWS_SECRET_ACCESS_KEY
to the following installation files (address #49 ):

* install_bootstrap
* boardwalk/dev.yml
* boardwalk/prod.yml
* boardwalk/conf/boardwalk.config.template

so they can be read into container dcc_dashboard-service_1. This is needed for upload to an S3 bucket. Installation succeeded and correctly added the environment variables to `.env`.

### CODE REVIEW

File: boardwalk/conf/boardwalk.config.template [hunk 1]
```diff
@@ -1,3 +1,6 @@
+azul_s3_bucket={{ AZUL_S3_BUCKET }}
+user_aws_key_id={{ AWS_ACCESS_KEY_ID }}
+user_aws_secret_key={{ AWS_SECRET_ACCESS_KEY }}
```
Comment (coverbeck) [inline boardwalk/conf/boardwalk.config.template:3]: I think you should match the names how it already is in the develop branch. That way when we eventually reconcile the two branches, it will be less confusing.

https://github.com/DataBiosphere/cgp-deployment/blob/develop/boardwalk/conf/boardwalk.config.template#L22

File: boardwalk/dev.yml [hunk 2]
```diff
@@ -4,6 +4,9 @@ services:
   dcc-dashboard-service:
     environment:
       APACHE_PATH: "${apache_path}"
+      AWS_ACCESS_KEY_ID: "${user_aws_key_id}"
```
Comment (coverbeck) [inline boardwalk/dev.yml]: Same as previous comment.

File: install_bootstrap [hunk 3]
```diff
@@ -249,17 +248,36 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
 #            read -ep $'What is the email address for notification?  Previous value: \n' -i "$EMAIL_ADDRESS" email_address
 #         else
 #            read -ep $'What is the email address for notification? \n' email_address
-#      fi
+      #      fi
```
Comment (coverbeck) [inline install_bootstrap]: Please undo this change.

File: install_bootstrap [hunk 4]
```diff
@@ -46,6 +46,7 @@ function check_for_repo {
     fi
 }
 
+
```
Comment (coverbeck) [inline install_bootstrap]: Please undo this change

File: install_bootstrap [hunk 5]
```diff
@@ -249,17 +248,36 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
...
+      
       dcc_dashboard_host='dcc_dashboard_host'
       ask_question "What is the hostname of dcc-dashboard? (This is likely the domain name resolving to your virtual machine.)" "$DCC_DASHBOARD_HOST" "DCC Dashboard Host" $dcc_dashboard_host
-        
+      
+      ############################
+      # AWS questions            #
+      ############################
+
+      user_aws_key_id='user_aws_key_id'
```
Comment (coverbeck) [inline install_bootstrap]: Same as comment about matching develop

File: install_bootstrap [hunk 6]
```diff
@@ -249,17 +248,36 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
...
+      ask_question "What is your AWS Key?" "$AWS_ACCESS_KEY_ID" "AWS Key name" $user_aws_key_id
+
+      user_aws_secret_key='user_aws_secret_key'
```
Comment (coverbeck) [inline install_bootstrap]: Same as comment about matching develop

File: install_bootstrap [hunk 7]
```diff
@@ -275,6 +293,10 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
       fc_lambda_domain='fc_lambda_domain'
       ask_question "What is the domain of the bagit-firecloud-lambda?" "$FC_LAMBDA_DOMAIN" "FireCloud Bagit Lambda Domain" $fc_lambda_domain
 
+      azul_s3_bucket='azul_s3_bucket'
+      ask_question "What is the name of the AWS S3 bucket to store BDBags?" "$AZUL_S3_BUCKET" "Azul S3 Bucket" $azul_s3_bucket
```
Comment (coverbeck) [inline install_bootstrap:296]: 👍

File: install_bootstrap [hunk 8]
```diff
@@ -314,29 +336,32 @@ while [[ "${run_boardwalk^^}" != 'Y' &&  "${run_boardwalk^^}" != 'N' ]] ; do
 
       cat > boardwalk_launcher_config/boardwalk.config <<CONFIG
 {
-"GOOGLE_CLIENT_ID":"${google_client_id}",
-"GOOGLE_CLIENT_SECRET":"${google_client_secret}",
-"GOOGLE_SITE_VERIFICATION_CODE":"${google_site_verification_code}",
+"AZUL_S3_BUCKET":"${azul_s3_bucket}",
```
Comment (coverbeck) [inline install_bootstrap:338]: Same comment about matching develop

**Review by coverbeck — CHANGES_REQUESTED** [review]
Were you going to update the readme as part of this PR? I think you should.

File: boardwalk/conf/boardwalk.config.template (hunk 1 above)
Comment (mjkrause) [reply boardwalk/conf/boardwalk.config.template:3]: I actually followed the naming in `develop`, but I started with `install_bootstrap`.

https://github.com/DataBiosphere/cgp-deployment/blob/develop/install_bootstrap#L230
https://github.com/DataBiosphere/cgp-deployment/blob/develop/install_bootstrap#L397

The problem is that in `develop` the same variable is named differently in different files. In fact, some of the variable names change within `install_bootstrap`, depending for what service the user is queried for. For example, compare these two lines:

https://github.com/DataBiosphere/cgp-deployment/blob/develop/install_bootstrap#L397
https://github.com/DataBiosphere/cgp-deployment/blob/develop/install_bootstrap#L942

I'm not sure if this is by design (in which case I'm missing the intention / utility)? I can use the names from the `boardwalk.config.template`. I had to make a decision whether to consistently name them in as in `install_bootstrap` or as in the mustache template.

Comment (coverbeck) [reply boardwalk/conf/boardwalk.config.template:3]: You're right. I think the variable name is intentionally different than the environment variable name on purpose. The way you have it is good.

Sorry that I sent you down the wrong path on this.

I think you should just copy the code as it is in develop branch.

**Review by coverbeck — CHANGES_REQUESTED** [review]
Looks good! Except you still need to update the Collection Information section of the README.md

File: boardwalk/conf/boardwalk.config.template (hunk 1 above)
Comment (mjkrause) [reply boardwalk/conf/boardwalk.config.template:3]: * I made the names for the AWS access key ID and secret access key consistent across all files.
* Amended the README.

**Review by coverbeck — CHANGES_REQUESTED** [review]
Looks good, except for one small change -- please mention the AZUL_S3_BUCKET in the Collecting Information section -- that it should refer to the bucket you describe how to create earlier in the readme.

Thanks.

**Review by coverbeck — APPROVED** [review]
